kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/KafkaService.java

```diff
@@ -786,6 +790,32 @@ private InternalClusterMetrics calculateClusterMetrics(
         });
   }
 
+  public Mono<Map<Integer, Map<String, DescribeLogDirsResponse.LogDirInfo>>> getClusterLogDirs(
+      KafkaCluster cluster, List<Integer> reqBrokers) {
+    return getOrCreateAdminClient(cluster)
+        .map(admin -> {
+          List<Integer> brokers = new ArrayList<>(cluster.getBrokers());
+          if (reqBrokers != null && !reqBrokers.isEmpty()) {
+            brokers.retainAll(reqBrokers);
+          }
+          return admin.getAdminClient().describeLogDirs(brokers);
+        })
+        .flatMap(result -> Mono.fromSupplier(() -> {
+          try {
+            return result.all().get();
```

Comment:
this one is dirty, you should wrap future into mono, please check util classes

Reply:
@germanosin I don't a better way. Mono has methods for CompletableFuture but not for Future (KafkaFuture in this case). If you have a better solution please advise. One thing I can do is move this code to util class and make this particular place cleaner.

Reply:
Sure, please use this method https://github.com/provectus/kafka-ui/blob/master/kafka-ui-api/src/main/java/com/provectus/kafka/ui/util/ClusterUtil.java#L57

Reply:
Thank you. Fixed.